src/main/java/xyz/nucleoid/mineout/game/map/MineoutCheckpoint.java

```diff
@@ -49,8 +64,31 @@ public void spawnPlayer(ServerPlayerEntity player, float rotation) {
         this.applyTo(player);
     }
 
+    public static void spawnFirework(@NotNull ServerWorld world, double x, double y, double z, int[] colors, boolean silent, int lifetime)
+    {
+        ItemStack fireworkStack = new ItemStack(Items.FIREWORK_ROCKET);
+
+        CompoundTag tag = fireworkStack.getOrCreateSubTag("Fireworks");
+        tag.putByte("Flight", (byte) 0);
+
+        ListTag explosions = new ListTag();
+        CompoundTag explosion = new CompoundTag();
+        explosion.putByte("Type", (byte) 1);
+        explosion.putIntArray("Colors", colors);
+        explosions.add(explosion);
+        tag.put("Explosions", explosions);
+
+        FireworkRocketEntity firework = new FireworkRocketEntity(world, x, y, z, fireworkStack);
+        firework.setSilent(silent);
+        if (lifetime >= 0)
+            ((FireworkRocketEntityAccessor) firework).setLifeTime(lifetime);
+        world.spawnEntity(firework);
+    }
+
     public void sendTaskTo(ServerPlayerEntity player) {
         if (this.task != null) {
+            spawnFirework(player.getServerWorld(), player.getX() + 0.25D, player.getY() + 0.25D, player.getZ() + 0.25D, new int[]{6719955, 2651799, 4312372, 14602026}, false, 0);
```

Comment:
The spawn position at +.25 seems a bit odd? Is there any reason for that specifically?
Also- can the colour codes be converted to hex for readability?

Reply:
+.25 is because it spawns in the block under the players feet without the 0.25D

Reply:
I'll push a update that will convert the hex to int

Reply:
Oh, I see. I mainly mean for the x/z since it's not centered

Reply:
If centered, it gets in the way of the players camera and blocks their line of sight.

Reply:
This can still happen- it just depends on which direction the player is looking. Maybe it should spawn above the player?

Reply:
I've made it so it spawns just above the feet. The particles are only partially visible to the player.

Reply:
If they're looking down, obviously it will get in their face. But I doubt people will look down as they'll be focused on the minigame